core/src/main/java/org/pentaho/reporting/platform/plugin/output/util/ExportReportUtils.java

```diff
@@ -0,0 +1,196 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.reporting.platform.plugin.output.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.pentaho.metadata.query.model.Query;
+import org.pentaho.pms.core.exception.PentahoMetadataException;
+import org.pentaho.reporting.engine.classic.core.Element;
+import org.pentaho.reporting.engine.classic.core.MasterReport;
+import org.pentaho.reporting.engine.classic.core.ReportDataFactoryException;
+import org.pentaho.reporting.engine.classic.core.ReportHeader;
+import org.pentaho.reporting.engine.classic.core.elementfactory.LabelElementFactory;
+import org.pentaho.reporting.engine.classic.core.parameters.DefaultParameterContext;
+import org.pentaho.reporting.engine.classic.core.parameters.ParameterDefinitionEntry;
+import org.pentaho.reporting.libraries.formula.EvaluationException;
+import org.pentaho.reporting.libraries.formula.parser.ParseException;
+
+import java.util.Date;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public interface ExportReportUtils {
+  String FILTERS_SUMMARY = "Filters Summary";
+  String PROMPTS_SUMMARY = "Prompts Summary";
+  String NO_FILTERS = "No filters used";
+  String NO_PROMPTS = "No prompts used";
+
+  // ISO_LOCAL_DATE is yyyy-MM-dd
+  DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ISO_LOCAL_DATE;
+
+  Map<Class<?>, Function<Object, String>> typeFormattingStrategies = Map.of(
```

Comment:
Rename typeFormattingStrategies to uppercase and mark static final

Reply:
Addressed